qml/FileListPage.qml

```diff
@@ -23,26 +23,59 @@ import Sailfish.Office.Files 1.0
 
 Page {
     id: page
-    property alias model: listView.model;
+    property alias model: filteredModel.sourceModel;
     property string title: "";
     property QtObject provider;
 
     allowedOrientations: Orientation.All;
 
+    FilterModel {
+        id: filteredModel
+        filterRegExp: RegExp("")
+    }
+
     SilicaListView {
         id: listView;
         anchors.fill: parent
 
         currentIndex: -1;
+        model: filteredModel
 
         children: ScrollDecorator { }
-        header: PageHeader {
+        header: SearchPageHeader {
+            id: header
+            width: parent.width
+
             //: Application title
             //% "Documents"
             title: qsTrId("sailfish-office-he-apptitle");
-
             // TODO: uncomment once there are more document sources
             // title: page.title;
+
+            //: Document search field placeholder text
+            //% "Search documents"
+            placeholderText: qsTrId("sailfish-office-tf-search-documents")
```

Comment:
I you prefer the text to be moved, since atm there's no use to have it here and it just generates extra unnecessary code. As for the header stuff that looks fine.